src/nemos/basis.py

```diff
@@ -1183,6 +1265,27 @@ class MultiplicativeBasis(Basis):
     n_basis_funcs : int
         Number of basis functions.
 
+    Examples
+    --------
+    # Generate sample data
+    >>> import numpy as np
+    >>> import nemos as nmo
+    >>> X, y = np.random.normal(size=(30, 3)), np.random.poisson(size=30)
+
+    # define two basis and multiply
+    >>> basis_1 = nmo.basis.BSplineBasis(10)
+    >>> basis_2 = nmo.basis.RaisedCosineBasisLinear(15)
+    >>> multiplicative_basis = nmo.basis.MultiplicativeBasis(basis1=basis_1, basis2=basis_2)
+    >>> transformed_X = multiplicative_basis.to_transformer().transform(X[:, 0:2])
+    >>> print(transformed_X.shape)
+    (30, 150)
+
+    # can multiply or add another basis to the AdditiveBasis object
```

Comment:
warn that the number of output features of the basis grows as the product of the n_features of the 1d basis. if you have 10, 15 and 100, the final basis will output 15000, as you are showing